### PR TITLE
Set commit tx to published when `OracleAttestedPriorCetTimelock`

### PR DESCRIPTION
**Pull request by da-kami**
Whenever we are in `PendingCommit` the link should be available already.

### CODE REVIEW

File: daemon/src/projection.rs [hunk 1]
```diff
@@ -494,6 +494,7 @@ impl Cfd {
                 self.aggregated.timelocked_cet = Some(timelocked_cet);
                 self.closing_price = Some(price);
 
+                self.aggregated.commit_published = true;
```
Comment (luckysori) [inline daemon/src/projection.rs:497]: 👍 I think we can add this to `RefundTimelockExpired` too. Also should it also be part of the `CetTimelockExpiredPostOracleAttestation` and `CetTimelockExpiredPriorOracleAttestation`?

**Review by luckysori — APPROVED** [review]
Makes sense to me.

**Review by bonomat — APPROVED** [review]
👍

**Review by klochowicz — APPROVED** [review]
👍

File: daemon/src/projection.rs (hunk 1 above)
Comment (da-kami) [reply daemon/src/projection.rs:497]: I would only set it in scenarios that are *directly* related to publishing `commit`. That is at least what we have been doing so far. 
`RefundTimelockExpired` sets `self.aggregated.refund_published = true;` because it is about the refund tx and only indirectly implies `commit`. But `RefundTimelockExpired` implies `CommitConfirmed` which sets `self.aggregated.commit_published = true;`.

I thought about `CetTimelockExpiredPostOracleAttestation` and `CetTimelockExpiredPriorOracleAttestation` as well, but they also don't direcly relate to publishing `commit` - they imply `CommitConfirmed` (timelock expiry can only happen if the tx made it on chain...). 
For me publishing `commit` is only directly related to `ManualCommit`, `OracleAttestedPriorCetTimelock` (in a case where `ManualCommit` is not triggered). Additionally `CommitConfirmed` is relevant for cases where the other party published `commit`. These three cases are covered now.
Note that `OracleAttestedPostCetTimelock` implies `CommitConfirmed` as well.

Comment (da-kami) [reply daemon/src/projection.rs:497]: We *could* opt for covering cases less strict, but that will make it hard to reason about the code I feel.

Comment (luckysori) [reply daemon/src/projection.rs:497]: I'm okay with your reasoning, so no need to change anything.

I find this whole `match` statement hard to follow, but I don't have a better idea a this point.

**Comment by da-kami** [comment]
bors ping

**Comment by da-kami** [comment]
bors retry

**Comment by da-kami** [comment]
bors retry

**Comment by da-kami** [comment]
bors retry
